tools/src/main/scala/scala/scalanative/build/NativeLib.scala

```diff
@@ -198,7 +203,7 @@ private[scalanative] object NativeLib {
             .stripSuffix(jarExt)
         NativeLib(
           src = path,
-          dest = workDir.resolve(s"$nativeCodePrefix-$name-$index")
+          dest = workDir.resolve(s"$nativeCodePrefix-$name-$confHash-$index")
```

Comment:
The other issue here is fact that `dest` dir would not be removed when it's no longer needed. So after frequent `changeConfig; reload;nativeLink` loops we would end up with garbage in the target dir. 

The nativeConfig hash might be a good start, but maybe we should store it in some file in the workdir directory as a form of persistent store between sbt reloads, and check if the current hash has changes. If so, we should perform a cleanup of workdir from all previous artifacts.

Reply:
>The nativeConfig hash might be a good start, but maybe we should store it in some file in the workdir directory as a form of persistent store between sbt reloads, and check if the current hash has changes. If so, we should perform a cleanup of workdir from all previous artifacts.

I think this might be a good idea.

Reply:
> The other issue here is fact that dest dir would not be removed when it's no longer needed. 

Not actually, they will be deleted when they are no longer needed


```scala
      Files
         // List all the files / dirs in `workDir` (somewhere under `target/scala-$version/native`)
        .list(workDir)

        // List dirs that contains native-lib NIRs such as `native-code-clib_native...`
        .filter(_.getFileName().toString() matches nativeCodePattern)

        // # List stale native-lib dirs that has different name than than current `workDir.resolve(s"$nativeCodePrefix-$name-$confHash-$index"`
        //(if `confHash` has different value than previous build, the stale dir will alive here)
        .filter(p => !expectedPaths.contains(p.toAbsolutePath())) 

        // Delete all stale directories
        .forEach(IO.deleteRecursive(_)) 
```
https://github.com/scala-native/scala-native/blob/bb8aef7288bbf1e89a7f2108aa50c8ef236a0b23/tools/src/main/scala/scala/scalanative/build/NativeLib.scala#L215-L219